src/browser/api/file_download.ts

```diff
@@ -0,0 +1,125 @@
+import { app } from 'electron';
+import { Identity } from '../../shapes';
+import { writeToLog } from '../log';
+import ofEvents from '../of_events';
+import route from '../../common/route';
+import * as coreState from '../core_state';
+
+const EVENT_TOPIC = 'window';
+export const FILE_DOWNLOAD_EVENTS = {
+    STARTED: 'file-download-started',
+    PROGRESS: 'file-download-progress',
+    COMPLETED: 'file-download-completed'
+};
+
+type DownloadState = 'started' | 'completed' | ' cancelled' |  'interrupted' | 'paused' | 'interrupted' | 'paused' | 'progressing';
+
+interface FileEvent {
+    fileUuid: string;
+    url: string;
+    mimeType: string;
+    fileName: string;
+    totalBytes: number;
+    startTime: number;
+    contentDisposition: string;
+    lastModifiedTime: string;
+    eTag: string;
+    topic: string;
+    uuid: string;
+    name: string;
+}
+
+interface FileDownloadLocation {
+    fileUuid: string;
+    path: string;
+    identity: Identity;
+}
+
+
+export const downloadLocationMap: Map<string, FileDownloadLocation> = new Map();
+
+//determines if the given identity should have access to the file
+export function hasAccess(identity: Identity, fileUuid: string): boolean {
+    const fileDownload = downloadLocationMap.get(fileUuid);
+    const requestorAnc = coreState.getAppAncestor(identity.uuid);
+    const fileOwnerAnc = fileDownload ? coreState.getAppAncestor(fileDownload.identity.uuid) : void 0;
+    if ((requestorAnc && fileOwnerAnc) && (requestorAnc.uuid === fileOwnerAnc.uuid)) {
+        return true;
```

Comment:
so we are being less restrictive than same uuid here? if created by a uuid, you can access their downloads?